@page "/login"
@inject NavigationManager NavigationManager
@inject IdentityAuthStateProvider AuthStateProvider

<AuthorizeView>
    <Authorized>
        <div class="alert alert-info mx-auto my-auto text-center">
            Знаете, конечно, похвально, что вы хотите регаться снова и снова и снова, но блин...<br />
            Хотя бы для приличия разлогиньтесь, что ли.
        </div>
    </Authorized>
    <NotAuthorized>
        <section class="card card-default mx-auto my-auto">
            @if (ProcessingQuery)
            {
                <StandbyMessage />
            }
            else
            {
                <h4 class="card-header">Вход</h4>
                <label class="text-center text-danger">@Error</label>
                <EditForm class="card-body" Context="LoginFormContext" Model="@User" OnValidSubmit="@LogIn">
                    <DataAnnotationsValidator />
                    <ValidationSummary />
                    <div class="form-group">
                        <label for="login">Логин</label>
                        <InputText @bind-Value="User.Login" class="form-control" id="login" placeholder="Логин" />
                    </div>
                    <div class="form-group">
                        <label for="password">Пароль</label>
                        <InputText @bind-Value="User.Password" class="form-control" id="password" placeholder="******" type="password" />
                    </div>
                    <input class="btn btn-primary" type="submit" value="Войти" />
                </EditForm>
            }
        </section>
    </NotAuthorized>
</AuthorizeView>

@code {
    private string Error { get; set; }
    private bool ProcessingQuery { get; set; }
    private LoginParams User { get; set; } = new LoginParams();

    private async void LogIn()
    {
        Error = null;
        ProcessingQuery = true;
        try
        {
            await AuthStateProvider.Login(User);
            NavigationManager.NavigateTo("");
        }
        catch (NotAuthenticatedException)
        {
            Error = "Неверные имя пользователя или пароль";
        }
        catch (Exception ex)
        {
            Error = "Похоже, нет соединения с сервером";
            Console.WriteLine(ex.Message);
        }
        ProcessingQuery = false;
        StateHasChanged();
    }
}
